packit_service/models.py

```diff
@@ -1546,27 +1562,56 @@ def get_all_by(
         All owner/project_name builds sorted from latest to oldest
         with the given commit_sha and optional target.
         """
-        non_none_args = {
-            arg: value for arg, value in locals().items() if value is not None
-        }
-
-        return (
+        query = (
             sa_session()
             .query(CoprBuildTargetModel)
-            .filter_by(**non_none_args)
+            .join(
+                CoprBuildTargetModel.group_of_targets,
+            )
+            .join(
+                PipelineModel,
+                PipelineModel.copr_build_group_id == CoprBuildGroupModel.id,
+            )
+            .join(
+                ProjectEventModel,
+                PipelineModel.project_event_id == ProjectEventModel.id,
+            )
+            .filter(CoprBuildTargetModel.project_name == project_name)
+            .filter(ProjectEventModel.commit_sha == commit_sha)
```

Comment:
Are we ok with having lower performance here? (And likely in every other build/test model class)